pkg/cluster/kubeclient/roundtrip.go

```diff
@@ -46,5 +49,16 @@ func NewRestConfig(log *logrus.Entry, config *v1.Config, cs *api.OpenShiftManage
 
 	restconfig.WrapTransport = roundtrippers.NewRetryingRoundTripper(log, cs.Location, cs.Properties.NetworkProfile.PrivateEndpoint, disableKeepAlives)
 
+	if cs.Properties.NetworkProfile.PrivateEndpoint != nil {
+		restconfig.Dial = func(ctx context.Context, network, addr string) (net.Conn, error) {
```

Comment:
This code moved as a knock-on effect of the revendor to origin's kubernetes tree.